src/core/contracts/src/main/scala/Params.scala

```diff
@@ -135,3 +135,14 @@ trait HasEvaluationMetric extends Wrappable {
   /** @group setParam */
   def setEvaluationMetric(value: String): this.type = set(evaluationMetric, value)
 }
+
+trait HasValidationIndicatorCol extends Wrappable {
+  val validationIndicatorCol: Param[String] =
+    new Param[String](this, "validationIndicatorCol", "Indicates whether the row is for training or validation")
```

Comment:
add a nonempty check here instead of throughout the code

Reply:
it can be empty, it just means we don't have a validation dataset